lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -180,13 +182,16 @@ def script
           # Strip invalid architectures
           strip_invalid_archs() {
             binary="$1"
+            warn_missing_arch=${2:-true}
             # Get architectures for current target binary
             binary_archs="$(lipo -info "$binary" | rev | cut -d ':' -f1 | awk '{$1=$1;print}' | rev)"
             # Intersect them with the architectures we are building for
             intersected_archs="$(echo ${ARCHS[@]} ${binary_archs[@]} | tr ' ' '\\n' | sort | uniq -d)"
             # If there are no archs supported by this binary then warn the user
             if [[ -z "$intersected_archs" ]]; then
-              echo "warning: [CP] Vendored binary '$binary' contains architectures ($binary_archs) none of which match the current build architectures ($ARCHS)."
+              if [[ "$warn_missing_arch" == "true" ]]; then
+                echo "warning: [CP] Vendored binary '$binary' contains architectures ($binary_archs) none of which match the current build architectures ($ARCHS)."
```

Comment:
for xcframeworks, it's possible for a dSYM to be excluded completely due to archs not matching. this isn't an error, since there are multiple dSYMs instead of 1